extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpSecurityRecorder.java

```diff
@@ -233,6 +233,11 @@ public void handle(RoutingContext event) {
             if (authenticator == null) {
                 authenticator = CDI.current().select(HttpAuthenticator.class).get();
             }
+
+            // we put RoutingContext to duplicated context local data so that users don't have to activate CDI request
+            // context during authentication and authorization when proactive auth is enabled
+            HttpSecurityUtils.setRoutingContextAttribute(event);
+
             //we put the authenticator into the routing context so it can be used by other systems
             event.put(HttpAuthenticator.class.getName(), authenticator);
```

Comment:
Just out of curiosity: why using `Class::getName`?
If we have a serie of possible keys somewhere, I would put something there, with a specific name unrelated to the class's name, which can still bother the JVM to init it  (and cache it , see https://github.com/openjdk/jdk/blob/f55381950266088cc0284754b16663675867ac87/src/java.base/share/classes/java/lang/Class.java#L988C43-L988C43).

Reply:
I just wanted to follow pattern that users already know in Quarkus OIDC (for example) 

https://github.com/quarkusio/quarkus/blob/447067819a58a10841375f41c12c36fb0eef5dbf/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcUtils.java#L349

I'm fine with introducing String constant, but there can be some users already doing `RoutingContext.class.getName` elsewhere.

Reply:
This is a good point; so let's still with what's the expected usage than :+1: